irohad/model/model_crypto_provider_impl.hpp

```diff
@@ -31,6 +34,8 @@ namespace iroha {
 
       bool verify(const Query &query) const override;
 
+      bool verify(std::shared_ptr<Query> query) const;
```

Comment:
What is the purpose of checking `sh_p` query? Maybe it will be confusing for a reader because in fact `verify` not requires ownership.

Reply:
Sorry, it came from deep Thursday's midnight )